integration_tests/gimbal.cpp

```diff
@@ -82,6 +85,78 @@ TEST_F(SitlTest, GimbalTakeoffAndMove)
     std::this_thread::sleep_for(std::chrono::seconds(3));
 }
 
+TEST_F(SitlTest, GimbalROIOffboard)
+{
+    DroneCore dc;
+
+    ConnectionResult ret = dc.add_udp_connection();
+    ASSERT_EQ(ret, ConnectionResult::SUCCESS);
+
+    // Wait for device to connect via heartbeat.
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    Device &device = dc.device();
+    auto telemetry = std::make_shared<Telemetry>(device);
+    auto gimbal = std::make_shared<Gimbal>(device);
+    auto action = std::make_shared<Action>(device);
+    auto offboard = std::make_shared<Offboard>(device);
+
+    while (!telemetry->health_all_ok()) {
+        LogInfo() << "waiting for device to be ready";
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    const Telemetry::Position &position = telemetry->position();
+
+    // set the ROI location: a bit to the north of the vehicle's location
+    const double latitude_offset_deg = 3. / 111111.; // this is about 3 m
+    send_gimbal_roi_location(gimbal, position.latitude_deg + latitude_offset_deg,
+                             position.longitude_deg, position.absolute_altitude_m + 1.f);
+
+    action->arm();
```

Comment:
Its good to handle API result.

Reply:
Right. I didn't do it here because: the other test in this file doesn't do it either, and I wanted to avoid code cluttering so it's easier to read. If desired, I can add the checks.

Reply:
I'd just put it inside an `EXPECT_EQ`, or even `ASSERT_EQ` because the rest of the test won't make sense without arming and takeoff.

Reply:
Ok, done.